dotnet/samples/GettingStartedWithAgents/BedrockAgent/Step07_BedrockAgent_Declarative.cs

```diff
@@ -192,9 +210,12 @@ private async Task InvokeAgentAsync(Agent agent, string input)
         }
         finally
         {
-            var bedrockAgent = agent as BedrockAgent;
-            Assert.NotNull(bedrockAgent);
-            await bedrockAgent.Client.DeleteAgentAsync(new() { AgentId = bedrockAgent.Id });
+            if (deleteAgent)
+            {
+                var bedrockAgent = agent as BedrockAgent;
+                Assert.NotNull(bedrockAgent);
+                await bedrockAgent.Client.DeleteAgentAsync(new() { AgentId = bedrockAgent.Id });
+            }
```

Comment:
Suggest removing `Assert` as this is a Demostration intended to be run in a Console application.

```suggestion
            if (deleteAgent && deleteAgent is BedrockAgent bedRockAgent)
            {
                await bedrockAgent.Client.DeleteAgentAsync(new() { AgentId = bedrockAgent.Id });
            }
```